core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -302,6 +317,43 @@ func (c *coordinator) getBlocks(
 	return
 }
 
+// getBlockhashesMapping returns the blockhashes within a range.
+func (c *coordinator) getBlockhashesMapping(
+	ctx context.Context,
+	startHeight uint64,
+	endHeight uint64,
+) (blockhashesMapping map[uint64]common.Hash, err error) {
+
+	blockHeights := make([]uint64, endHeight-startHeight+1)
+	for i := range blockHeights {
+		blockHeights[i] = uint64(i) + startHeight
+	}
+
+	heads, err := c.lp.GetBlocks(blockHeights, pg.WithParentCtx(ctx))
+	if len(heads) != len(blockHeights) {
+		err = fmt.Errorf("could not find all heads in db: want %d got %d", len(blockHeights), len(heads))
+		return
+	}
+
+	blockhashesMapping = make(map[uint64]common.Hash)
+	for _, head := range heads {
+		blockhashesMapping[uint64(head.BlockNumber)] = head.BlockHash
+	}
+	return
+}
+
+func removeDuplicates(uintSlice []uint64) []uint64 {
+	allKeys := make(map[uint64]bool)
```

Comment:
Nit: Idiomatic way to represent a set in Go is `map[T]struct{}` rather than `map[T]bool`.